src/binary/ibc-setup/commands/init.ts

```diff
@@ -0,0 +1,57 @@
+import fs from 'fs';
+import path from 'path';
+
+import { Command } from 'commander';
+import yaml from 'js-yaml';
+
+import { GlobalOptions } from '../types';
+import { resolveOption } from '../utils/resolve-option';
+
+type Options = GlobalOptions & {
+  src: string;
+  dest: string;
+};
+
+type ProgramOptions = Partial<Options>;
+
+function getDefaultHome() {
+  if (!process.env.HOME) {
+    throw new Error('$HOME environment variable is not set.');
+  }
+
+  return `${process.env.HOME}/.ibc-setup`;
+}
+
+const REGISTRY_FILE = 'registry.yaml';
+
+export function init(_: unknown, program: Command) {
+  const programOptions: ProgramOptions = program.opts();
+
+  const options = {
+    src: resolveOption('src', programOptions.src, process.env.RELAYER_SRC),
+    dest: resolveOption('dest', programOptions.dest, process.env.RELAYER_DEST),
+    home: resolveOption('home', programOptions.home, getDefaultHome),
+  };
+
+  run(options);
+}
+
+export function run(options: Options) {
+  if (!fs.existsSync(options.home)) {
+    fs.mkdirSync(options.home, { recursive: true });
+    console.log(`Initialized home directory at ${options.home}`);
+  } else if (!fs.lstatSync(options.home).isDirectory()) {
+    throw new Error(`${options.home} must be a directory. It is a file.`);
+  }
+
+  const REGISTRY_FILE_PATH = path.join(options.home, REGISTRY_FILE);
+  if (!fs.existsSync(REGISTRY_FILE_PATH)) {
+    // TODO: download registry.yaml from default location
+  } else if (!fs.lstatSync(REGISTRY_FILE_PATH).isFile()) {
+    throw new Error(`${REGISTRY_FILE_PATH} must be a file. It is a directory.`);
+  }
+
+  const registry = yaml.load(fs.readFileSync(REGISTRY_FILE_PATH, 'utf-8'));
```

Comment:
This will throw if the file was removed while running the program or it has no appropriate permissions to be read or if yarm parsing fails. However, I think it's okay to not catch this since it produces a meaningful error and we want it to throw. Maybe later if we want to log this and re-throw?

Reply:
Sounds good. We can throw any non-recoverable errors.

When we integrate the logger, I would propose wrapping all the "run" functions with something that catches any error and passes it to `log.error` before `process.exit(1)`. But that is a different issue

Reply:
>When we integrate the logger, I would propose wrapping all the "run" functions with something that catches any error and passes it to log.error before process.exit(1)

Yeah, an error boundary would be nice!